### DISCUSSION TITLE
Have ddns-updater running on Android (yes this is borderline)

### DISCUSSION THREAD

**Discussion by edaubert1kmapied** [qdm12/ddns-updater#575] [Q&A] [answered]
Hi,

I am currently trying to run ddns-updater on an old android phone using termux.

To do so, I chose to use a dedicated fully qualified name to access it and so as I use ddns-updater on all my other servers, I taught of doing the same here.

Even if my phone is rooted, I choose not to bother about docker because I don't want to rebuild a kernel.
So I get the docker image for arm64 architecture and extract `/updater/app` as it is a golang program.

```shell
docker pull --platform linux/arm64 qmcgaw/ddns-updater:latest
docker export $(docker create --platform linux/arm64 qmcgaw/ddns-updater:latest)  -o ddns-updater-arm64.tgz
tar -xvf ddns-updater-arm64.tgz updater/app
```

So now some details about my network configuration :
My ISP provider give me a partial dual IPv4/IPv6 stack.
Basically I can't use all the available ports on my IPv4 stack as it is shared with some other clients of my ISP providers
And I can fully use my IPv6 stack so everything I run is on IPv6 as well as my DNS.

However when I run ddns-updater on my android phone I got issues about resolving some DNS which don't have IPv6 counterparts (github and ovh api) while running a nslookup get me a IPv4 address.
Additionally ddns-updater got my IPv6 address when requesting it on `https://api6.ipify.org` for example so I am guessing the DNS is working for IPv6 website on ddns-updater.

Here is some partial logs I got from ddns-updater when running it on termux
```
WARN performing request: Get "https://github.com": dial tcp: lookup github.com on [::1]:53: read udp [::1]:43532->[::1]:53: read: connection refused
...
ERROR obtain adjusted time from OVH: doing http request: Get "https://eu.api.ovh.com/1.0/auth/time": dial tcp: lookup eu.api.ovh.com on [::1]:53: read udp [::1]:60847->[::1]:53: read: connection refused
```

I know this is a very borderline usage and many things could have an impact so I guess android is messing with me (or termux maybe).

Anyway if something has an idea about my issue, fell free to help me :stuck_out_tongue_winking_eye:

**Comment by edaubert** [comment, marked as answer]
Ok I found the root cause of my issue.

Android doesn't give a `/etc/resolv.conf`. Termux gives one but the file is located on `/data/data/com.termux/files/usr/etc/resolv.conf` and so ddns-updater doesn't use it.

There is two solution :
- compiling the app on termux to be correctly linked to the resolv.conf
- use `proot` and specify a correct resolv.conf

I chose the second one : 
`proot -b $PREFIX/etc/resolv.conf:/etc/resolv.conf $HOME/ddns/ddns-updater` with `$PREFIX=/data/data/com.termux/files/usr/` (and given by termux itself)

More info on https://stackoverflow.com/questions/38959067/dns-lookup-issue-when-running-my-go-app-in-termux

**Reply by qdm12** [reply]
Hi there, thanks for sharing! Note two things you could try:

- set `RESOLVER_ADDRESS=1.1.1.1` (or `--resolver-address`) so it doesn't read /etc/resolv.conf etc.
- there are now pre-built binaries for each releases (starting with v2.5.0) so you could use that for your cpu architecture

If you have any idea how to improve the program/code for your use case, feel free to share it and I can implement it 😉 !
